test/unit/eval_algorithms/test_faithfulness.py

```diff
@@ -0,0 +1,413 @@
+        output = faithfulness_algo.evaluate(
+            dataset_config=dataset_config,
+            num_records=162,
+            save=True,
+        )
+        # THEN
+        mock_build_pipeline.assert_called_with(
+            mock_default_judge_model,
+            LONG_FORM_ANSWER_PROMPT,
+            NLI_STATEMENTS_MESSAGE,
+        )
+        mock_evaluate_dataset.assert_called_once_with(
+            dataset=mock_dataset,
+            pipeline=mock_build_pipeline.return_value,
+            dataset_name=dataset_config.dataset_name,
+            eval_name=faithfulness_algo.eval_name,
+            metric_names=["faithfulness"],
+            eval_results_path="/path/to/results",
+            agg_method=MEAN,
+            save=True,
+            save_strategy=None,
+        )
+        assert output == [mock_evaluate_dataset.return_value]
+
+    class TestCaseFaithfulnessEvaluate(NamedTuple):
+        input_dataset: Dataset
+        dataset_config: Optional[DataConfig]
+        statements_output: List[str]
+        verdicts_output: List[str]
+        expected_response: List[EvalOutput]
+
+    @pytest.mark.parametrize(
+        "test_case",
+        [
+            #
+            TestCaseFaithfulnessEvaluate(
+                input_dataset=DATASET_WITH_CONTEXT,
+                dataset_config=DataConfig(
+                    dataset_name="my_custom_dataset",
+                    dataset_uri="tba",
+                    dataset_mime_type=MIME_TYPE_JSON,
+                    model_input_location="tba",
+                    target_output_location="tba",
+                    model_output_location=None,
+                    category_location="tba",
+                ),
+                statements_output="Here are the statements created from the given answer:\nStatement: Einstein was born in Germany.\nStatement: Einstein was born on 20th March 1879.",
+                verdicts_output='here are the verdicts for the statements:\n1. statement: einstein was born in germany.\nexplanation: the context states that einstein was "a german-born theoretical physicist". this supports that he was born in germany.\nverdict: yes\n2. statement: einstein was born on 20th march 1879.  \nexplanation: the context states that einstein was "born 14 march 1879". this contradicts the statement that he was born on 20th march 1879.\nverdict: no\nfinal verdicts in order:\nyes. no.',
+                expected_response=[
+                    EvalOutput(
+                        eval_name="faithfulness",
+                        dataset_name="my_custom_dataset",
+                        prompt_template=None,
+                        dataset_scores=DATASET_SCORES,
+                        output_path="/tmp/eval_results/faithfulness_my_custom_dataset.jsonl",
+                    )
+                ],
+            ),
+        ],
+    )
+    @patch("fmeval.eval_algorithms.faithfulness.get_dataset")
+    def test_faithfulness_evaluate(self, mock_get_dataset, test_case):
```

Comment:
Can we move this to `test/integration/` instead? We can add larger datasets as additional test cases in the future. Instead of using a mocked model runner, you can create a dummy model runner whose `predict` method is copied over from `predict_side_effect`.

Reply:
Will create integration tests in upcoming PRs.